### DISCUSSION TITLE
Panic: Buffer is empty

### DISCUSSION THREAD

**Discussion by wajidkhanp** [postmanlabs/postman-flows#232] [Defect]
I am getting this alert [ Panic: Buffer is Empty] when I try to run requests with 4 APIS is flow with ForEach having 4 variables in the list.

Developer logs are as below. Please help. 

Flow works fines with a single use and when we automate with Foreach and list of variables its fails

**Comment by wajidkhanp** [comment]
electron/js2c/renderer_init.js:13 (electron) The remote module is deprecated. Use https://github.com/electron/remote instead.
log @ electron/js2c/renderer_init.js:13
vendor-ui.js:5735 Initializing the global config
vendor-ui.js:5943 MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 model-events listeners added. Use emitter.setMaxListeners() to increase limit
    at d (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2960278)
    at a.addListener (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2962862)
    at d.u (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5735:423113)
    at Object.init (file:///Applications/Postman.app/Contents/Resources/app.asar/js/requester.js:459:166645)
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/requester.js:67:16330
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1609667
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1609011
    at h (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1607077)
    at m (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1606995)
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1609296
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:1609644
    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/requester.js:67:20193
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Proxy process identifier proxy_node_process_prod
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 App updater event Object
vendor-ui.js:5943 App updater event Object
vendor-ui.js:5943 App updater event Object
vendor-ui.js:5943 App updater event Object
vendor-ui.js:5943 Disconnected from Interceptor
10vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Successfully received user organizations Object
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@758] TypeError: Cannot read property 'unsubscribe' of undefined
    at RequesterModalContainer.js:115
    at X (vendor-ui.js:2)
    at n (vendor-ui.js:2)
    at g (vendor-ui.js:2)
    at e.onInvalidate (vendor-ui.js:2)
    at e.runReaction (vendor-ui.js:2)
    at ze (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at vendor-ui.js:2
    at ll (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at Be (vendor-ui.js:2)
    at Ne (vendor-ui.js:2)
    at Module.Ft (vendor-ui.js:2)
    at Object.populateBase (vendor-ui.js:2985)
    at Object.<anonymous> (vendor-ui.js:2985)
    at Generator.next (<anonymous>)
    at s (vendor-ui.js:2985)
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 Warning: a promise was created in a handler but was not returned from it    at M (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4895623)    at I.r._warn (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4892337)    at Object.checkForgottenReturns (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4901285)    at I._settlePromiseFromHandler (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4924160)From previous event:    at I.O [as _captureStackTrace] (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4895295)    at I._then (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4919858)    at I.then (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5943:4918228)    at s.hydrate (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:7577:61940)    at X (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2838524)    at s.n (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2838299)    at new r (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5822:771129)    at new s (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:7577:60018)    at s (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5822:964358)    at new r (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:7577:105794)    at s (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5822:964358)    at new r (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:5812:64532)    at _i (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3217560)    at zs (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3234978)    at _a (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3273041)    at bl (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3264515)    at fl (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3264440)    at al (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3261470)    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3213218    at t.unstable_runWithPriority (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3617281)    at Yo (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3212927)    at Jo (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3213163)    at Ko (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3213098)    at ll (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:3261761)    at Fe (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2851215)    at Be (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2850798)    at Ne (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2847479)    at Ft (file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2:2860313)    at file:///Applications/Postman.app/Contents/Resources/app.asar/js/vendor-ui.js:2965:103824
(anonymous) @ vendor-ui.js:5943
3vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 sign in bailed. force connect.
vendor-ui.js:5943 Skipping force reconnect as still preparing and no attempt of connection has been made yet.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
238Error: <circle> attribute cx: Unexpected end of attribute. Expected length, "".
238Error: <circle> attribute cy: Unexpected end of attribute. Expected length, "".
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
3vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 environment:update:21819509-0176-c3fc-e846-680ec2eb90e6
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@12733] TypeError: Cannot read property 'isActive' of undefined
    at vendor-ui.js:5822
    at vendor-ui.js:2
    at fe (vendor-ui.js:2)
    at e.track (vendor-ui.js:2)
    at g (vendor-ui.js:2)
    at e.onInvalidate (vendor-ui.js:2)
    at e.runReaction (vendor-ui.js:2)
    at ze (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at vendor-ui.js:2
    at ll (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at Be (vendor-ui.js:2)
    at Ne (vendor-ui.js:2)
    at Ft (vendor-ui.js:2)
    at vendor-ui.js:5822
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
clientstream.launchdarkly.com/eval/60645109ece3b60c64ac5e02/eyJjdXN0b20iOnsiYXV0b2ZsZXgiOmZhbHNlLCJ0ZWFtSWQiOiIxMTU5OTQiLCJwbGF0Zm9ybSI6ImRlc2t0b3AiLCJhcHBWZXJzaW9uIjoiOS4xMS4wIiwicmVsZWFzZUNoYW5uZWwiOiJwcm9kIiwidXNlckNyZWF0ZWRBdCI6MTUzODAxMzg4ODAwMCwiZGF5c1NpbmNlVXNlckNyZWF0ZSI6MTIxOCwidGVhbVNpemUiOjE1MCwicm9sZXMiOlsidXNlciJdLCJ0ZWFtQ3JlYXRlZEF0IjoxNTMwMTY2MTU4MDAwLCJkYXlzU2luY2VUZWFtQ3JlYXRlIjoxMzA5fSwia2V5IjoiNTQ2MTQwMSIsInVzZXJJZCI6IjU0NjE0MDEifQ:1 Failed to load resource: net::ERR_NETWORK_CHANGED
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
clientstream.launchdarkly.com/eval/60645109ece3b60c64ac5e02/eyJjdXN0b20iOnsiYXV0b2ZsZXgiOmZhbHNlLCJ0ZWFtSWQiOiIxMTU5OTQiLCJwbGF0Zm9ybSI6ImRlc2t0b3AiLCJhcHBWZXJzaW9uIjoiOS4xMS4wIiwicmVsZWFzZUNoYW5uZWwiOiJwcm9kIiwidXNlckNyZWF0ZWRBdCI6MTUzODAxMzg4ODAwMCwiZGF5c1NpbmNlVXNlckNyZWF0ZSI6MTIxOCwidGVhbVNpemUiOjE1MCwicm9sZXMiOlsidXNlciJdLCJ0ZWFtQ3JlYXRlZEF0IjoxNTMwMTY2MTU4MDAwLCJkYXlzU2luY2VUZWFtQ3JlYXRlIjoxMzA5fSwia2V5IjoiNTQ2MTQwMSIsInVzZXJJZCI6IjU0NjE0MDEifQ:1 Failed to load resource: net::ERR_NETWORK_CHANGED
vendor-ui.js:5943 TypeError: Cannot destructure property 'body' of 'e' as it is undefined.
    at vendor-ui.js:4713
    at Generator.next (<anonymous>)
    at s (vendor-ui.js:4713)
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Error: Failed to execute the request, check Console for more information.
    at vendor-ui.js:4713
    at Generator.next (<anonymous>)
    at s (vendor-ui.js:4713)
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 Could not find on r. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 walkCollectionTree could not find the request 45dc1f03-df04-4e4b-aee7-7ac779f58d72 in  Object
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 globals:update:206392c9-d9bf-477d-8e98-4897b8bfdc00
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@12644] TypeError: Cannot read property 'isActive' of undefined
    at vendor-ui.js:5822
    at vendor-ui.js:2
    at fe (vendor-ui.js:2)
    at e.track (vendor-ui.js:2)
    at g (vendor-ui.js:2)
    at e.onInvalidate (vendor-ui.js:2)
    at e.runReaction (vendor-ui.js:2)
    at ze (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at vendor-ui.js:2
    at ll (vendor-ui.js:2)
    at Fe (vendor-ui.js:2)
    at Be (vendor-ui.js:2)
    at Ne (vendor-ui.js:2)
    at Ft (vendor-ui.js:2)
    at vendor-ui.js:5822
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
app.getpostman.com/api/public_profile:1 Failed to load resource: the server responded with a status of 400 ()
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
5vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 globals:update:206392c9-d9bf-477d-8e98-4897b8bfdc00
11vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
5vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 Uncaught TypeError: Cannot set property 'pos' of undefined
    at vendor-ui.js:4704
    at b.setState (requester.js:62)
    at X (vendor-ui.js:2)
    at b.n (vendor-ui.js:2)
    at Z.nodeMoved (vendor-ui.js:4704)
    at X (vendor-ui.js:2)
    at Z.n (vendor-ui.js:2)
    at vendor-ui.js:4556
    at f (vendor-ui.js:5943)
    at v (vendor-ui.js:5943)
    at y (vendor-ui.js:5943)
    at r (vendor-ui.js:5943)
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
2vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
2vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
app.getpostman.com/api/public_profile:1 Failed to load resource: the server responded with a status of 400 ()
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
vendor-ui.js:5943 Uncaught TypeError: Cannot set property 'pos' of undefined
    at vendor-ui.js:4704
    at b.setState (requester.js:62)
    at X (vendor-ui.js:2)
    at b.n (vendor-ui.js:2)
    at Z.nodeMoved (vendor-ui.js:4704)
    at X (vendor-ui.js:2)
    at Z.n (vendor-ui.js:2)
    at vendor-ui.js:4556
    at f (vendor-ui.js:5943)
    at v (vendor-ui.js:5943)
    at y (vendor-ui.js:5943)
    at r (vendor-ui.js:5943)
sentry-app.postmanlabs.com/api/22/store/?sentry_key=8fb2d5d1749642eface793d3dd93a6d8&sentry_version=7:1 Failed to load resource: the server responded with a status of 429 (TOO MANY REQUESTS)
7vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
5vendor-ui.js:5943 The React Flow parent container needs a width and a height to render the graph.
(anonymous) @ vendor-ui.js:5943
2vendor-ui.js:5943 Could not find on l. Missing or invalid parameter.
(anonymous) @ vendor-ui.js:5943
9vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.
(anonymous) @ vendor-ui.js:5943
4vendor-ui.js:5943 LD: [warn] Error on stream connection: {"isTrusted":true}, will continue retrying every 1000 milliseconds.

**Comment by saswatds** [comment]
@wajidkhanp Can you give us a quick look into what your flow looks like?

**Comment by PericlesFeltrin** [comment]
I am testing flows and received the same error.
My test return 2 items, but the foreach iterate only in first item.

Part of flow:
![image](https://user-images.githubusercontent.com/5816150/152077903-5ff0a7ce-2dff-4d98-b11b-c3175ecdfd1a.png)

Test Summary 
![image](https://user-images.githubusercontent.com/5816150/152077717-6db80325-a21a-4f2d-89a9-e3a398f24d78.png)

My test return 2 items, but the foreach iterate only in first item.

Output in terminal
![image](https://user-images.githubusercontent.com/5816150/152077673-dc95a5bf-e105-42ca-9e61-dcec9d4aa8de.png)

**Reply by saswatds** [reply]
Thanks @PericlesFeltrin for the flow. I will take a deep look into this. Seems like a issue with the internal execution implementation.
